TACC/Core-CMS-Custom
# WP-738 Hook up Edit Registration and Renew Registration to Registration Form.

## Overview
With RegistrationForm available, hookup edit registration and renew registration to it.
Reuse as much of the component code as possible.

## Related
- [WP-738](https://tacc-main.atlassian.net/browse/WP-738)

## Changes

* Make RegistrationForm adaptable to edit and also modal 
* Move the registration form submission to a `usePost` with `mutation` to allow invalidating the registration listing when an edit is done.
* Do not load all the registration modal content for listing page. Only load on demand. This is both a perf improvement and also to keep data in sync between updates.
* Update submitter listing
* Manage registration listing to allow renew for nonAdmin scenario and update for admin scenario. (different actions).
* Adjust the registration form to handle reg_id form url and make a request.


## Testing

1. Go to http://localhost:8000/administration/list-registration-requests/ and use edit action.
   * edit registration
   * add contact
   * add entitiy
   * remove contact and add contact
   * remove entity.
   * check if the updates are seen if edit action is opened again. 
   * check if the registration level updates are available in the listing and 'view' action
2. Go to http://localhost:8000/register/list-registration-requests/ and use renew action

## UI
Admin edit: 
<img width="1106" alt="Screenshot 2024-11-04 at 3 09 17 PM" src="https://github.com/user-attachments/assets/89aec1cc-2d4f-4b21-a444-3ad0cfc9dd07">

Submitter Listing: 
<img width="1139" alt="Screenshot 2024-11-04 at 3 07 58 PM" src="https://github.com/user-attachments/assets/cf8d3d8c-cdf8-402d-b5c5-6f1daa4f6902">

## Notes
* This request is still not available on server side: http://localhost:8000/register/request-to-submit/?reg_id=2
